src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private Node<T> next;
+        private T value;
+
+        Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<T>(tail, value, null);
+        Node<T> temp = tail;
+        tail = newNode;
+        if (temp == null) {
+            head = newNode;
+        } else {
+            temp.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            checkPositionIndex(index);
+            Node<T> foundNode = node(index);
+            Node<T> prev = foundNode.prev;
+            Node<T> newNode = new Node<T>(prev, value, foundNode);
+            foundNode.prev = newNode;
+            if (prev == null) {
+                head = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> foundNode = node(index);
+        if (foundNode == null) {
+            return null;
```

Comment:
If you checked node index before it mean that you have this node in your list, how can this node == null?